### PR TITLE
implemented profile filter

### LINKED ISSUE #582: Add click event to profitable crafts
There should be more information about a craft when it is clicked.
Some things requested so far
* lowest bin of ingredients
* lowest bin of the resulting item
* crafting recipe

### LINKED ISSUE #583: Add profile select for profitable auctions
https://sky.coflnet.com/api/profile/{uuid} can be used to get a list of profiles for one player.
That id should be passed to the `https://sky.coflnet.com/api/craft/profit?profile={profileId}&player={uuid}`  
Every profile has a `cute_name` that should be displayed.
The default should be the profile with the `current` set to true

### PR DESCRIPTION
**Pull request by matthias-luger**
fixes #583
fixes #582

### CODE REVIEW

**Comment by matthias-luger** [comment]
@Ekwav did you try it? Does it work like you intended it to?

**Comment by matthias-luger** [comment]
@Ekwav ping

**Comment by matthias-luger** [comment]
@Ekwav please try again

**Comment by matthias-luger** [comment]
@Ekwav  pong

File: src/components/CraftsList/CraftsList.tsx [hunk 1]
```diff
@@ -15,27 +16,52 @@ export function CraftsList() {
...
 
     useEffect(() => {
+        setIsLoadingCrafts(true);
+        setIsLoadingProfileData(true);
         loadCrafts();
+        // eslint-disable-next-line react-hooks/exhaustive-deps
     }, []);
 
     function loadCrafts(playerId?: string, profileId?: string) {
         api.getProfitableCrafts(playerId, profileId).then(crafts => {
```
Comment (Ekwav) [inline src/components/CraftsList/CraftsList.tsx:32]: crafts are not cached via the profile id thus leading to data unexpectedly being not refreshed when a different profile is selectd

File: src/components/CraftsList/CraftsList.tsx [hunk 2]
```diff
@@ -102,20 +125,32 @@ export function CraftsList() {
...
-                    </div> : null
-            }
+            <div>
+                {
+                    isLoadingProfileData && isLoggedIn ?
+                        getLoadingElement() :
+                        <div>
+                            {
+                                !isLoggedIn ? <p>To use the the profile filter please login with Google and connect your Minecraft Account:</p> :
+                                    !accountInfo?.mcId ? <p>To use the the profile filter please connect your Minecraft Account</p> : ""
```
Comment (Ekwav) [inline src/components/CraftsList/CraftsList.tsx]: explain how to connect the account and what that means?

**Review by Ekwav — CHANGES_REQUESTED** [review]

**Review by Ekwav — APPROVED** [review]
LGTM expect for the missing explanation which I created an issue for

**Review by Ekwav — APPROVED** [review]
there is a merge conflict

**Pull request merged**